pkg/k8s/api.go

```diff
@@ -73,6 +74,22 @@ func (kubeAPI *KubernetesAPI) CheckVersion(versionInfo *version.Info) error {
 	return nil
 }
 
+func (kubeAPI *KubernetesAPI) CheckProxyVersion(pods []v1.Pod, version string) error {
+	for _, pod := range pods {
+		for _, container := range pod.Spec.Containers {
+			if container.Name == ProxyContainerName {
```

Comment:
Given that there can be a different image for the `init-container`, do we also need to check for the condition `if container.Name == InitContainerName`?

Reply:
Hmm, that's a good question. The way inject is currently implemented, somebody would need to hand-edit the YAML file to end up with different versions for the linkerd-proxy and linkerd-init containers. Given that, plus the fact that the linkerd-init container is no longer running when this check is executed, I'm inclined to only check the version of the running linkerd-proxy container, and have that suffice for the version check.

Also, related to your question, init containers aren't included in the slice of `pod.Spec.Containers`. We would instead need to look at `pod.Spec.InitContainers` if we wanted to get the state of the pod's init containers.

Reply:
Ahh that's right, InitContainers do run to completion, so the check is no longer necessary. This makes sense to me. 👍